### DISCUSSION TITLE
Cypress prevents axios to add XSRF token to cookies request header

### DISCUSSION THREAD

**Discussion by htmoia** [cypress-io/cypress#27414] [Questions and Help]
While testing our angular/axios application, cypress seems to prevent adding the XSRF-TOKEN to the cookies header while performing POST requests. 
This leads to 403 error sent from our spring boot backend which cannot compare the X-XSRF token with the cookie value.

While debugging with `cy.getCookie` and `cy.log` when running tests with `cypress open` we also realized the cookie values logged by cypress are different to the one cookie values shown in the chrome dev-tools.

Is this a known issue? How can we prevent cypress of removing cookie headers?

**Comment by portyu9** [comment]
Hi @htmoia 

Axios should not be adding the XSRF value to the `Cookie` header. There are two separate things involved:

- the browser sends the `XSRF-TOKEN` cookie
- Axios reads that cookie and sends its value as the `X-XSRF-TOKEN` header

So I would first verify which half is actually missing rather than assuming Cypress removed the cookie:

```js
cy.intercept('POST', '**/api/**').as('post')

cy.getCookie('XSRF-TOKEN').should('exist')

// trigger the POST here

cy.wait('@post')
  .its('request.headers')
  .should('have.property', 'x-xsrf-token')
```

Also check the cookie's `domain` and `path`. `cy.getCookie('XSRF-TOKEN')` defaults to the hostname of the current page, so comparing it with a similarly named cookie for another host/path in DevTools can give misleading results.

For a **cross-origin** frontend/API setup, current Axios separates the two settings:

```js
axios.create({
  withCredentials: true,
  withXSRFToken: true,
})
```

`withCredentials` allows the cross-site request to carry credentials/cookies; `withXSRFToken` tells current Axios to copy the XSRF cookie value into the XSRF header.

I would not manually construct a `Cookie` header in Cypress as the first fix. If the cookie exists but the outgoing `X-XSRF-TOKEN` header is absent, that points to the Axios/origin configuration. If the cookie itself is not being sent, then check its domain, path, `SameSite`, and `Secure` attributes.

There is not enough evidence in the post to conclude that Cypress itself is stripping this cookie.

Please mark this answer as accepted if it helped, thank you.
